packages/components/src/text/README.md

```diff
@@ -0,0 +1,18 @@
+# Text
+
+A text component for styling text.
+
+## Usage
+
+```jsx
+import {Text} from '@wordpress/components';
+
+const HeroPanel = () => (
+	<>
+		<Text variant="title.large" as="h1">Hello World!</Text>
```

Comment:
Shouldn't the `title.large` option default to `h1`? In general, it would be great to have this more opinionated so all popular HTML tags are covered with variants. `as` can be still used to customize it if necessary.

Reply:
We could potentially do that. Is the location where a `title.large` usually gets used usually a `h1` too? (not just thinking about Gutenberg but thinking about other plugins too)

If we change the default element for one variant, should we change the default element for the other variants too?

Reply:
> If we change the default element for one variant, should we change the default element for the other variants too?

Yes, this is what I meant by saying – it would be great to have this more opinionated so all popular HTML tags are covered with variants. I assume that the whole idea is to stop using HTML tags, so this should make it easier to keep the code platform agnostic.

Reply:
👍I'll let @davewhitley jump in to discuss whether there's a clear link between each variant and and a semantic element.

Reply:
It wasn't designed with a heading level in mind, no. I'm hesitant to assign each heading level to a typographic style, because they should be separate. Each typographic style can use whatever HTML element is appropriate. 

If we default to a `p` element for each style we are forcing the user of the component to make a choice on which heading level element to use, which is a good thing imo. 

I'm afraid that eventually people will just associate `h1` for Title Large, just as an example.

I don't know enough about the Text component or how/when it should be used to give an authoritative answer though. I do know that I'm concerned about people ignoring the heading level and just using the default value, which will result in inaccessible markup. I'm also concerned about people choosing a specific typographic style _because_ of the heading level associated with it. E.g. "I'll choose Title Medium because I need a level 2 heading". It’s important to realize that the style of a heading is independent from the semantic underlying element.

Maybe look at Material UI or Polaris to see how it's done?

Reply:
I'm not sure Polaris is helpful...

The `Display Text` variants (kind of like our `title.*` variants) maps to a `p`.
The `Heading` variant which offers a single size maps to a `h2`.
The `Subtitle` variant which offers a single size maps to a `h3`.
Everything other variant maps to a `p`.

I'd advocate that we don't be overly opinionated by defaulting to anything other than a `p` at this time and _wait and see_ how the component gets used and whether there are any consistent usage patterns that form and be implemented into the component.